### PR TITLE
Read static CSP configuration values from file

### PR DESCRIPTION
**Pull request by patrickmann**
See #15370 for details

Initially you could specify a CSP string within the `@CSP` annotation. Combining this with a config file and dynamically updated auth hosts was messy. Now you specify a CSP "group" in the annotation, which corresponds to a group of values from the config file. 

Values are now stored in memory as sets, to avoid duplicates and simplify processing. Any dynamic update is applied to that data structure, so there is no ambiguity or on-the-fly parsing and concatenating of values.

### CODE REVIEW

File: graylog2-server/src/main/resources/org/graylog2/security/csp.config [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+# value is not ignored, and is treated as part of the value.
+#
+# Example:
+# default.img-src=my_imgsrc.com
+# default.connect-src=my_connectsrc.com:9999
+
+default.default-src='self'
+default.style-src='self' 'unsafe-inline'
+default.script-src='strict-dynamic' 'nonce-{nonce}' 'unsafe-eval'
+default.img-src=data:*
```
Comment (dennisoelkers) [inline graylog2-server/src/main/resources/org/graylog2/security/csp.config:30]: This should be `data:` instead and we should also include `'self'` and the exception for OSM here:
```suggestion
default.img-src=data: 'self' https://*.tile.openstreetmap.org
```

File: graylog2-server/src/main/resources/org/graylog2/security/csp.config [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+swagger.style-src='self' 'unsafe-inline'
+swagger.script-src='self' 'unsafe-eval' 'unsafe-inline'
+swagger.img-src=*
```
Comment (dennisoelkers) [inline graylog2-server/src/main/resources/org/graylog2/security/csp.config:34]: Is swagger actually loading images from somewhere else than `'self'`?

**Review by dennisoelkers — CHANGES_REQUESTED** [review]
Looks good so far, added on suggestion and a question inline.

File: graylog2-server/src/main/resources/org/graylog2/security/csp.config (hunk 1 above)
Comment (patrickmann) [reply graylog2-server/src/main/resources/org/graylog2/security/csp.config]: updated

File: graylog2-server/src/main/resources/org/graylog2/security/csp.config (hunk 2 above)
Comment (patrickmann) [reply graylog2-server/src/main/resources/org/graylog2/security/csp.config]: Doesn't appear to - works fine with `'self'`

**Comment by bernd** [comment]
@patrickmann @dennisoelkers I noticed we are implementing a custom property file parser for the CSP config file. What's the reason for configuring the CSP feature using a property-style file and a custom parser instead of a JSON or YAML file?

**Comment by patrickmann** [comment]
> What's the reason for configuring the CSP feature using a property-style file and a custom parser instead of a JSON or YAML file?

A Java property file seemed the simplest solution. Do we have a coding convention not to use these?

Allowing multiple entries for the same attribute is a convenience feature; but it also means it is no longer a valid property file.

**Comment by bernd** [comment]
> A Java property file seemed the simplest solution. Do we have a coding convention not to use these?
> 
> Allowing multiple entries for the same attribute is a convenience feature; but it also means it is no longer a valid property file.

We don't have a coding convention for not using property files. It's about having to implement a custom parser. :slightly_smiling_face: If we would use JSON/YAML for that, we wouldn't have to implement and maintain a custom property file parser. Or we don't use the convenience feature and a regular property parser.

**Review by dennisoelkers — APPROVED** [review]
Looks great, works fine! Thanks a lot!

**Pull request merged**